litmus-portal/authentication/api/routes/user_router.go

```diff
@@ -0,0 +1,184 @@
+		if err != nil {
+			log.Warn(err)
+			c.JSON(utils.ErrorStatusCodes[utils.ErrInvalidCredentials], presenter.CreateErrorResponse(utils.ErrInvalidCredentials))
+			return
+		}
+		token, err := authenticatedUser.GetSignedJWT()
+		if err != nil {
+			log.Error(err)
+			c.JSON(utils.ErrorStatusCodes[utils.ErrServerError], presenter.CreateErrorResponse(utils.ErrServerError))
+			return
+		}
+		c.JSON(200, gin.H{
+			"access_token": token,
+			"expires_in":   86400,
+			"type":         "Bearer",
+		})
+	}
+}
+
+func updatePassword(service user.Service) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		username := c.MustGet("username").(string)
+		uid := c.MustGet("uid").(string)
+		var adminUser entities.User
+		adminUser.UserName = username
+		var userPasswordRequest entities.UserPassword
+		err := c.BindJSON(&userPasswordRequest)
+		userPasswordRequest.Username = username
+		adminUser.ID, _ = primitive.ObjectIDFromHex(uid)
+		if err != nil {
+			log.Warn(err)
+			c.JSON(utils.ErrorStatusCodes[utils.ErrInvalidRequest], presenter.CreateErrorResponse(utils.ErrInvalidRequest))
+			return
+		}
+		if userPasswordRequest.OldPassword == "" || userPasswordRequest.NewPassword == "" {
+			c.JSON(utils.ErrorStatusCodes[utils.ErrInvalidRequest], presenter.CreateErrorResponse(utils.ErrInvalidRequest))
+			return
+		}
+		err = service.UpdatePassword(&userPasswordRequest, true)
+		if err != nil {
+			log.Info(err)
+			c.JSON(utils.ErrorStatusCodes[utils.ErrInvalidCredentials], presenter.CreateErrorResponse(utils.ErrInvalidCredentials))
+			return
+		}
+		c.JSON(200, gin.H{
+			"message": "password has been reset",
+		})
+	}
+}
+
+func resetPassword(service user.Service) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		uid := c.MustGet("uid").(string)
+		var adminUser entities.User
+		adminUser.UserName = c.MustGet("username").(string)
+		adminUser.ID, _ = primitive.ObjectIDFromHex(uid)
+		var userPasswordRequest entities.UserPassword
+		err := c.BindJSON(&userPasswordRequest)
+		if err != nil {
+			log.Info(err)
+			c.JSON(utils.ErrorStatusCodes[utils.ErrInvalidRequest], presenter.CreateErrorResponse(utils.ErrInvalidRequest))
+			return
+		}
+		if userPasswordRequest.Username == "" || userPasswordRequest.NewPassword == "" {
+			log.Warn(err)
+			c.JSON(utils.ErrorStatusCodes[utils.ErrInvalidRequest], presenter.CreateErrorResponse(utils.ErrInvalidRequest))
+			return
+		}
+		err = service.IsAdministrator(&adminUser)
+		if err != nil {
+			log.Info(err)
+			c.AbortWithStatusJSON(utils.ErrorStatusCodes[utils.ErrUnauthorised], presenter.CreateErrorResponse(utils.ErrUnauthorised))
+			return
+		}
+		err = service.UpdatePassword(&userPasswordRequest, false)
+		if err != nil {
+			log.Error(err)
+			c.AbortWithStatusJSON(utils.ErrorStatusCodes[utils.ErrServerError], presenter.CreateErrorResponse(utils.ErrServerError))
+			return
+		}
+		c.JSON(200, gin.H{
+			"message": "password has been reset successfully",
```

Comment:
Typo: suddenly -> successfully

Reply:
I do not seem to find the mentioned typo anywhere 😅

Reply:
![Screenshot from 2021-06-09 09-57-57](https://user-images.githubusercontent.com/45846277/121293454-4554b500-c909-11eb-8d5f-8ac8de377098.png)

IDK how its happening, but my system autocorrects successful-ly to suddenly :sweat_smile: